workshop/content/intermediate/operations/resource-importing/_index.md

```diff
@@ -0,0 +1,341 @@
+
+```yaml
+Parameters:
+  LatestAmiId:
+    Type: AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>
+    Default: /aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2
+    Description: Fetching the latest AMI ID for Amazon Linux
+  InstanceType:
+    Description: Select the type of the EC2 instance.
+    Type: String
+    AllowedValues:
+      - t2.nano
+      - t2.micro
+      - t2.small
+```
+
+```yaml
+  Instance:
+    DeletionPolicy: Retain
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !Ref LatestAmiId
+      InstanceType: !Ref InstanceType
+      Tags:
+        - Key: Name
+          Value: InstanceImport
+```
+
+4. Save the template file. Update the stack again with the updated `resource-import-challenge.yaml` template, which has no parameters section and the instance resource definition. This will remove the instance from the stack but will not delete it because of `DeletionPolicy` set to `Retain`.
+5. After this stack update, add the two removed code blocks from Step 3 back to the `resource-import-challenge.yaml` template, and save it.
+6. Select the stack named `resource-import-challenge` and from **Stack actions**, choose **Import resources into stack.**
+7. Read the **Import Overview** and choose **Next.**
+8. From **Specify template**, choose **Upload a template file**. Upload the `resource-import-challenge.yaml` template saved for this lab part and choose **Next.**
+9. For [**Identifier Value**](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import.html#resource-import-overview), use the **InstanceId** which you noted from EC2 console in the first part of this challenge.
+10. For parameters, select **t2.micro** for the instance type. Here you are matching the actual configuration on the instance which is **t2.micro instance type**.
+11. Choose **Import Resources** in the next page.
+
+{{% /expand %}}
+
+
+
+You can find the template for the solution in the `code/solutions/resource-importing/resource-import-challenge-solution.yaml` example template.
+
+Great work! You have now learned how to match the CloudFormation stack configuration with the actual configuration on the resource when there is an out of band change.
+
+**Use Cases:**
+
+1. You created an AWS resource (for example S3 bucket) manually 2 years ago and you would like to manage it using AWS CloudFormation.
+2. You want to bring certain types of resources into a single stack for easier management (for example, IAM Role resources).
+3. You have two separate stacks, and you want to bring them into a single stack by nesting stacks. Refer to this [document](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import-nested-stacks.html) for nesting an existing stack.
+4. You want to match the CloudFormation configuration for a resource which was updated out of band.
+
+### Cleanup
+
+Choose to follow cleanup steps shown next to cleanup resources you created with this lab:
+
+
```

Comment:
Add a preliminary step where you direct the user in updating each of the 3 templates and remove`DeletionPolicy` attributes, and updating relevant stacks first.  Otherwise, removing stacks as they are now will retain resources.

Reply:
Added steps for removing Deletion Policy by updating the stack and then cleaning them up.